specification/resources/resource-manager/Microsoft.Resources/stable/2020-06-01/resources.json

```diff
@@ -6031,6 +6031,10 @@
     "WhatIfOperationProperties": {
       "description": "Deployment operation properties.",
       "properties": {
+        "correlationId": {
```

Comment:
@njuCZ @Tiano2017 WhatIfOperationProperties is a GET only property so adding a new property to the response will not cause any trouble to the GET-PUT pipeline. Can we not consider this as a breaking change?

Reply:
@majastrz can you help take a look at this? Would appreciate hearing your thoughts as well.

Reply:
The general versioning guidelines are at https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md. Because it's a stable API version, I think we will have to bump the API version ☹.

Reply:
> Even though we recommend clients ignore new fields, there are many libraries and clients that fail when new fields are introduced.

Yeah this is something we must to consider. I'll abandon the PR and bump the API version then. We will also have to re-record the tests for the azure cli module again, which is a nightmare 🤦‍♂️...